### PR TITLE
Add types for HistGradientBoosting max_features argument

### LINKED ISSUE #322: Missing parameter for sklearn.ensemble.HistGradientBoostingRegressor
Since scikit-learn 1.4 `HistGradientBoostingRegressor` has had parameter `max_features`: https://github.com/scikit-learn/scikit-learn/blob/6e9039160/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py#L1499

This appears to be missing from the stubs: https://github.com/microsoft/python-type-stubs/blob/main/stubs/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.pyi#L90

### PR DESCRIPTION
**Pull request by ara-incom**
This pull request adds type annotation for the max_features argument of the scikit-learn histogram gradient boosting classes: BaseHistGradientBoosting, HistGradientBoostingRegressor, and HistGradientBoostingClassifier (all implemented in https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py)

Fixes #322

### CODE REVIEW

**Review by debonte — APPROVED** [review]

**Comment by debonte** [comment]
@ara-incom, thanks for the contribution!
